src/components/DatasetListing/DatasetListing.js

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import { Box } from '@mui/material';
+import { useWorkspaceMainDatasets } from '../../hooks/WorkspaceDatasetsHooks';
+import { useUserEmail } from '../../state/auth/hooks';
+import { ListingEmpty, ListingHeader } from '../ListingData';
+import { DatasetListingTable } from './components/DatasetListingTable';
+
+export const DatasetListing = ({
+  onCreateDataset,
+  onEditDataset,
+  onCopyDataset,
+  onShareDataset,
+  onDeleteDataset,
+  datasets: datasetsProp,
+}) => {
+  const datasetsFromHook = useWorkspaceMainDatasets() ?? [];
+  const datasets = datasetsProp ?? datasetsFromHook;
+  const userEmail = useUserEmail();
+  const isEmpty = datasets.length === 0;
+  const { t } = useTranslation();
+
+  const userName = React.useMemo(() => {
+    if (userEmail) {
+      const parts = userEmail.split('@')[0].split('.');
+      if (parts.length >= 2) {
+        return parts.map((p) => p.charAt(0).toUpperCase() + p.slice(1)).join(' ');
+      }
+      return userEmail.split('@')[0];
+    }
+    return 'Anonymous';
+  }, [userEmail]);
```

Comment:
The user name should already be stored in redux: you can use the hook `useUserName` instead of `useUserEmail`